torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -244,6 +246,10 @@ const Tensor& resize_(
     AT_ERROR("cannot resize variables that has a forward grad");
   }
 
+  // If `self` was resized, increment the version.
+  if (org_size != size) {
```

Comment:
It would be better to have this in an `InplaceOrView` key kernel like we do for `copy_`.
This code will not work for inference mode.

Reply:
Ah, will have a follow-up PR. Thanks!